core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,1002 @@
+                            if (allow_non_zero_integ_part is False and
+                            rule_spec.inputs['x'] != 0):
+                                fraction_interaction_invalid_values.append(
+                                    'The rule {rule} of answer '.format(
+                                        rule=rule_spec_index) +
+                                    'group {answer_group} has '.format(
+                                        answer_group=ans_group_index) +
+                                    'non zero integer part having ' +
+                                    'rule type HasIntegerPartEqualTo.')
+
+                # Validates various values of NumericInput interaction.
+                if value.interaction.id == 'NumericInput':
+                    for rule_spec in answer_group.rule_specs:
+                        rule_spec_index = str(answer_group.rule_specs.index(
+                            rule_spec))
+                        # Validates if tol value < 0 of rule IsWithTol.
+                        if rule_spec.rule_type == 'IsWithinTolerance':
+                            if rule_spec.inputs['tol'] < 0:
+                                numeric_input_interaction_values.append(
+                                    'The rule {rule} of answer '.format(
+                                        rule=rule_spec_index) +
+                                    'group {answer_group} having '.format(
+                                        answer_group=ans_group_index) +
+                                    'rule type IsWithinTolerance ' +
+                                    'have tol value less than zero.')
+
+                        # Validates if the 1st value is greater than 2nd.
+                        if rule_spec.rule_type == 'IsInclusivelyBetween':
+                            if rule_spec.inputs['a'] > rule_spec.inputs['b']:
+                                numeric_input_interaction_values.append(
+                                    'The rule {rule} of answer '.format(
+                                        rule=rule_spec_index) +
+                                    'group {answer_group} having '.format(
+                                        answer_group=ans_group_index) +
+                                    'rule type IsInclusivelyBetween ' +
+                                    'have a value greater than b value')
+
+                # Validates for NumberWithUnits interaction.
+                if value.interaction.id == 'NumberWithUnits':
+                    # Validates if IsEqui comes before IsEqual for same value.
+                    for rule_spec in answer_group.rule_specs:
+                        rule_spec_index = str(answer_group.rule_specs.index(
+                            rule_spec))
+                        if rule_spec.rule_type == 'IsEquivalentTo':
+                            number_with_units_rules.append(
+                                rule_spec.inputs['f'])
```

Comment:
This might work if it is only limited to the rule specs under a single answer group. Still, there might be a possibility that if it's present in a different answer group with the same value then it will be hard to track it with the boolean value.
Thanks

Reply:
Yeah okay, the for loop is so long that I forgot we are still inside it, maybe try to split it up more